cpp/CMakePresets.json

```diff
@@ -1,165 +1,231 @@
+      }
+    },
+    {
+      "name": "ninja-debug-cuda",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and CUDA integration",
+      "cacheVariables": {
+        "ARROW_CUDA": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-filesystems",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and filesystems",
+      "cacheVariables": {
+        "ARROW_GCS": "ON",
+        "ARROW_HDFS": "ON",
+        "ARROW_S3": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-flight",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and Flight",
+      "cacheVariables": {
+        "ARROW_FLIGHT": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-gandiva",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and Gandiva",
+      "cacheVariables": {
+        "ARROW_GANDIVA": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-python",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and Python support",
+      "cacheVariables": {
+        "ARROW_PYTHON": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-maximal",
+      "inherits": "ninja-debug",
+      "displayName": "Debug build with everything enabled (except benchmarks and CUDA)",
+      "cacheVariables": {
+        "ARROW_BUILD_EXAMPLES": "ON",
+        "ARROW_BUILD_UTILITIES": "ON",
+        "ARROW_FLIGHT": "ON",
+        "ARROW_GANDIVA": "ON",
+        "ARROW_GCS": "ON",
+        "ARROW_HDFS": "ON",
+        "ARROW_HIVESERVER2": "ON",
+        "ARROW_ORC": "ON",
+        "ARROW_PLASMA": "ON",
+        "ARROW_PYTHON": "ON",
+        "ARROW_S3": "ON",
+        "ARROW_SKYHOOK": "ON",
+        "ARROW_TENSORFLOW": "ON",
+        "PARQUET_BUILD_EXAMPLES": "ON",
+        "PARQUET_BUILD_EXECUTABLES": "ON",
+        "PARQUET_REQUIRE_ENCRYPTION": "ON"
+      }
+    },
+
+    {
+      "name": "ninja-release-minimal",
+      "inherits": "ninja-debug-minimal",
+      "displayName": "Release build without anything enabled",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
+      "name": "ninja-release-basic",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Release build with reduced dependencies",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
+      }
+    },
+    {
+      "name": "ninja-release",
+      "inherits": "ninja-debug",
+      "displayName": "Release build with more optional components",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
+        "ARROW_BUILD_TESTS": "OFF",
+        "ARROW_EXTRA_ERROR_CONTEXT": "OFF",
+        "CMAKE_BUILD_TYPE": "Release"
```

Comment:
We can have multiple inheritances using a list. Have you tried inheriting from `"ninja-debug, ninja-release-basic"` ?